docs: Http (#45)
NativeScript/docs: content/guide/core/http.md

---
title: Http
---
<!-- TODO: Add links -->

> This class allows you to make http requests.

Reviewer: `Http` provides utility methods to make all sorts of http networking requests.

> This class allows you to make http requests.

Reviewer: ```suggestion
Http provides utility methods to make all sorts of http networking requests.
```

## Using Http

### GET request with a response body as a string

A simple way to make a GET request returning a response as a string is to use the [getString()](#getstring) method:

```ts
Http.getString('https://httpbin.org/get').then(
  (result: string) => {
    // do something with the string response
  },
  e => {}
)
```

> ### GET request with a response body as a JSON

Reviewer: ```suggestion
### GET request with a response body as JSON
```

Use the [getJSON()](#getjson) method for a GET request with a response as a JSON:

```ts
Http.getJSON('https://httpbin.org/get').then(

>   (result: any) => {

Reviewer: ```suggestion
  (result) => {
```

---

#### headers

```ts
const requestOptions = {
 
  headers: { "header-name" : "header-value" },
}
```
_Optional_: gets or sets the request headers in JSON format.

---

#### content

```ts
const requestOptions = {
 
  content: formData,
}

//or

const requestOptions = {
 
  content: "some string",
}
```
_Optional_: gets or sets the request body.

---

#### timeout
```ts
const requestOptions = {
  timeout : 22333
}
```

_Optional_: gets or sets the request timeout in milliseconds.

---

#### dontFollowRedirects

```ts
const requestOptions = {
  dontFollowRedirects : true
}
```

_Optional_: boolean value that sets wether to **not** follow server's redirected request.

---

### HttpResponse interface

#### statusCode

Gets the response status code.

_Type_: `number`

---

#### content

Gets the response content.
_Type_: [HttpContent](#httpcontent-interface) 

---

#### headers

Gets the response headers

---

### HttpContent interface

#### raw

```ts
response.content?.raw
```

Gets the response body as raw data.

---

#### toString()

```ts
response.content?.toString()
```

Gets the response body as string.

---

#### toJSON()

```ts
response.content?.toJSON(encoding)
```

> Gets the response body as JSON object. The `encoding` is an optional of type `'UTF8' | 'GBK'`

Reviewer 2: We can keep `UTF8` & `GBK` as external links and on clicking, the CTA could be more detail/some docs page for encoding types

> Gets the response body as JSON object. The `encoding` is an optional of type `'UTF8' | 'GBK'`

Reviewer: ```suggestion
Gets the response body as JSON object. The `encoding` is optional of type `'UTF8' | 'GBK'`.
```
